src/mem/x86_64.rs

```diff
@@ -0,0 +1,69 @@
+use super::c_int;
+
+// On recent Intel processors, "rep movsb" and "rep stosb" have been enhanced to
```

Comment:
How does this implementation fare on non-intel implementations of x86_64?

Reply:
I've been investigating performance on AMD hardware (the only other x86 platform where anyone cares about performance). This has led me to modify the implementation. When I have some time, I'll post the results and update this comment to clarify the impact on AMD as well.

Reply:
On Intel fast string support is detectable by two flags in CPUID and one enable bit in IA32_MISC_ENABLE, if necessary. AMD may provide the same detection tools.

Reply:
So it looks like virtually all newish AMD and Intel processors support some sort of "REP MOVS enhancement" (i.e. rep movs is somehow better than a normal loop). However, if the ermsb feature flag isn't present (like on all AMD processors) then `rep movsq` seems better than `rep movsb`. With ermsb the two variants are about the same speed.

Given this, I just implemented the `rep movsq` version unconditionally without any CPUID checking. Variants that use `rep movsb` when Intel's ERMSB/FSRB feature is enabled could be added later, but there doesn't seem to be much of a gain (at least with the benchmarks I'm running here).